### PR TITLE
Consolidate text between ARGS SPEC and UG (part 1 of 3)

### PR DESCRIPTION
**Pull request by emlys**
# Description
First part of #567.
- Carbon
- Coastal Blue Carbon 
- Coastal Blue Carbon Preprocessor
- Coastal Vulnerability
- Crop Production Percentile
- Crop Production Regression
- DelineateIt
# Checklist
~- [ ] Updated HISTORY.rst (if these changes are user-facing)~
~- [ ] Updated the user's guide (if needed)~

### CODE REVIEW

File: src/natcap/invest/carbon.py
Comment (emlys) [reply src/natcap/invest/carbon.py:173]: Yea

File: tests/test_args_specs.py
Comment (emlys) [reply tests/test_args_specs.py:272]: yea, I was making sure to lowercase all the names. Removed this.

File: tests/test_args_specs.py
Comment (emlys) [reply tests/test_args_specs.py:52]: they were, but I don't think it's needed now.

File: src/natcap/invest/crop_production_regression.py
Comment (emlys) [reply src/natcap/invest/crop_production_regression.py:52]: I took this from text in the UG which said 12, but as far as I know this is still uncertain (#549)

File: src/natcap/invest/carbon.py
Comment (emlys) [reply src/natcap/invest/carbon.py:63]: > I almost wonder if we should have a LULC constant similar to the others

I like this

One advantage of a constant over a style guide is reducing repeated text - will reduce the number of words we need to get translated.

File: src/natcap/invest/carbon.py
Comment (emlys) [reply src/natcap/invest/carbon.py:92]: Following the 'spell out the first time' rule in this context may not be possible because the order of displaying these strings in the UG is not guaranteed to be the same. I think for common acronyms (LULC, DEM, AOI, TFA) it should be ok to define them only once for the whole user's guide (like on the data sources page). But also, sometimes it's helpful to write out.

File: src/natcap/invest/coastal_blue_carbon/coastal_blue_carbon.py
Comment (emlys) [reply src/natcap/invest/coastal_blue_carbon/coastal_blue_carbon.py:362]: I can see how this would be more annoying to edit in the future. For me this was less about standardizing phrasing and more about:
- reducing duplicated text to reduce translation time/costs
- ensuring we document the "condition" for conditionally-required args. most conditionally-required args simply depend on one other arg and we could auto-generate that description. but we allow the condition to be any python expression, and a few args use that feature. so auto-documenting is not really possible. I was hoping to get close to that but still allow a custom description for more complicated conditions.

There may not be a perfect solution here, I'm pretty ambivalent about it.

File: src/natcap/invest/carbon.py
Comment (emlys) [reply src/natcap/invest/carbon.py:48]: I was trying to use f strings as much as possible but this one wouldn't all fit in one line

File: src/natcap/invest/coastal_vulnerability.py
Comment (emlys) [reply src/natcap/invest/coastal_vulnerability.py:193]: Oh okay, that makes more sense.

File: src/natcap/invest/coastal_blue_carbon/coastal_blue_carbon.py
Comment (emlys) [reply src/natcap/invest/coastal_blue_carbon/coastal_blue_carbon.py:331]: Changed to `Run Valuation`

File: src/natcap/invest/coastal_blue_carbon/coastal_blue_carbon.py
Comment (emlys) [reply src/natcap/invest/coastal_blue_carbon/coastal_blue_carbon.py:354]: Right now the `REQUIRED_IF_SELECTED` is a standalone sentence ending with a period, so it wouldn't work exactly here.

File: src/natcap/invest/coastal_vulnerability.py
Comment (emlys) [reply src/natcap/invest/coastal_vulnerability.py:70]: Oh, I just moved that phrase over to the RST side... I'll take it out completely.

File: src/natcap/invest/coastal_vulnerability.py
Comment (emlys) [reply src/natcap/invest/coastal_vulnerability.py:159]: If 'linear unit' is too confusing, I'll just go with `u.none`. While any unit would work, you could even have a unitless "height index" and that would still work as long as higher points have bigger values.

File: src/natcap/invest/coastal_vulnerability.py
Comment (emlys) [reply src/natcap/invest/coastal_vulnerability.py:251]: I'd prefer not to make the `count` vs `none` distinction, since I don't fully understand it

File: src/natcap/invest/crop_production_regression.py
Comment (emlys) [reply src/natcap/invest/crop_production_regression.py:68]: thanks!

File: src/natcap/invest/carbon.py
Comment (emlys) [reply src/natcap/invest/carbon.py:173]: I changed this to "The relative annual increase". Hopefully that plus the type (ratio) will make it clear enough?

File: src/natcap/invest/carbon.py
Comment (emlys) [reply src/natcap/invest/carbon.py:63]: Thinking about this more, it probably won't help with translation. We can swap out translated text at the paragraph level, and probably sentences, but words/phrases need to be translated in context. So the benefit of using a constant is really just for consistency.

File: src/natcap/invest/coastal_blue_carbon/coastal_blue_carbon.py
Comment (dcdenu4) [reply src/natcap/invest/coastal_blue_carbon/coastal_blue_carbon.py:362]: > reducing duplicated text to reduce translation time/costs

I often forget about this part of it. It will take a little while to get in the mindset of being aware of the translation impact when working on future user guide changes.

File: src/natcap/invest/crop_production_regression.py
Comment (dcdenu4) [reply src/natcap/invest/crop_production_regression.py:52]: Ha! Apparently I forgot to "dig out" the conversation about this with Rich. Oops.

Comment (dcdenu4) [reply src/natcap/invest/crop_production_regression.py:52]: Okay, #549 now has the comment from Rich I was thinking of!

File: src/natcap/invest/carbon.py
Comment (emlys) [reply src/natcap/invest/carbon.py:173]: Actually I was wrong about this, not sure why I removed it.

File: src/natcap/invest/carbon.py
Comment (davemfish) [reply src/natcap/invest/carbon.py]: At least for me, TFA is not like those others in terms of the recognizability of the acronym.

File: src/natcap/invest/coastal_blue_carbon/coastal_blue_carbon.py
Comment (davemfish) [reply src/natcap/invest/coastal_blue_carbon/coastal_blue_carbon.py:362]: > > reducing duplicated text to reduce translation time/costs
> 
> I often forget about this part of it. It will take a little while to get in the mindset of being aware of the translation impact when working on future user guide changes.

It's true, but I wouldn't want us to go too far out of our way to conserve words. If we can afford the translation costs in general, then I'm sure we can afford a few extra words here and there.

File: src/natcap/invest/coastal_vulnerability.py
Comment (davemfish) [reply src/natcap/invest/coastal_vulnerability.py]: Sounds great to me

**Comment by emlys** [comment]
As we agreed in the coffee call this morning, I took out all the f-string variables and instead we'll follow a style guide for those things.

File: src/natcap/invest/crop_production_regression.py
Comment (emlys) [reply src/natcap/invest/crop_production_regression.py:52]: Oh thanks!! I updated this and also the UG page to say 10.

File: src/natcap/invest/carbon.py
Comment (emlys) [reply src/natcap/invest/carbon.py]: I'm going with `LULC` everywhere for now

**Review by dcdenu4 — APPROVED** [review]
Hey @emlys , thanks for addressing my comments. I think this is looking good.

**Review by davemfish — APPROVED** [review]
Latest changes look good to me!

**Review by davemfish — CHANGES_REQUESTED** [review]
Changes look good but a few tests are failing, I think related to these changes.

**Comment by emlys** [comment]
ReadTheDocs build is failing on GDAL install with this error: `error: Setup script exited with error in GDAL setup command: use_2to3 is invalid.`
probably because of [this setuptools bug](https://github.com/pypa/setuptools/issues/2775) released in the last couple days.

**Review by davemfish — APPROVED** [review]
Okay I approve! @emlys I'll let you decide whether to merge yet or hold out for anything else.

**Review by phargogh — APPROVED** [review]
Latest changes look great to me!  I'll leave it up to you @emlys about the readthedocs build and/or if there's anything else before merging.

**Comment by emlys** [comment]
The setuptools issue only happened because I forgot to update the GDAL requirement in `requirements-docs.yml` _and_ there was a change to setuptools. Because I had the GDAL version wrong, it was still trying to `pip install gdal==3.2.2` and so was running the GDAL setup. With the correct version, we `conda install` GDAL, so it doesn't matter for us if GDAL's setup is compatible with the setuptools version or not. 

Before: expected failure when the gdal library isn't already installed
```
$ conda install setuptools==57.4.0
$ pip install gdal==3.2.2
...
gdal_config_error: [Errno 2] No such file or directory: 'gdal-config'
    
Could not find gdal-config. Make sure you have installed the GDAL native library and development headers.
```

Now: different failure due to setuptools change
```
$ conda install setuptools==58.0.2
$ pip install gdal==3.2.2
...
error in GDAL setup command: use_2to3 is invalid.
WARNING: numpy not available!  Array support will not be enabled
```

**Pull request merged**
